### ISSUE TITLE
One Failing Test Case on Ubuntu

### ISSUE THREAD

**Issue by tiagomartinho** [apple/swift-nio#83]
### Insight
I saw that in the Docker configuration multiple versions of Swift are defined (4.0.2 & 4.0). There is a particular reason for it @tomerd?
Upgrading it to the latest version (4.0.3) makes all the Test Suite pass on Ubuntu. 
Should I do directly a PR with these changes?

### Expected behavior
All tests in the Test Suite to pass.

### Actual behavior
1 Failing Test Case:

> HTTPUpgradeTestCase.testDelayedUpgradeBehaviour

Logs:
```
test_1       | Test Case 'HTTPUpgradeTestCase.testDelayedUpgradeBehaviour' started at 2018-03-03 03:28:58.078
test_1       | /code/Tests/NIOHTTP1Tests/HTTPUpgradeTests.swift:694: error: HTTPUpgradeTestCase.testDelayedUpgradeBehaviour : XCTAssertFalse failed -
test_1       | Test Case 'HTTPUpgradeTestCase.testDelayedUpgradeBehaviour' failed (0.003 seconds)
```

### Steps to reproduce

1. Install Docker for Mac
2. Run: cd docker
2. Run: docker-compose up test

### If possible, minimal yet complete reproducer code (or URL to code)

Just follow the README instructions on how to run the test suite for Ubuntu using Docker.

### SwiftNIO version/commit hash

1.0.0 / 5ff6d9ba7241b1f4d8121d84531bf87c901d930b

### Swift & OS version (output of `swift --version && uname -a`)
Specified in the Docker configuration.

**Comment by tomerd** [comment]
thanks for reporting, will address shortly

**Comment by tiagomartinho** [comment]
@tomerd I open the PR for this issue https://github.com/apple/swift-nio/pull/89

**Comment by tomerd** [comment]
i'v determined that `HTTPUpgradeTestCase.testDelayedUpgradeBehaviour` is flaky, regardless of swift version. we'll fix that first then we can do the docker config cleanup and 4.0.3 upgrade

**Comment by Lukasa** [comment]
#92 should address the test failure.

**Comment by tomerd** [comment]
addressed via #92 and #90
